xla/service/gpu/priority_fusion.cc

```diff
@@ -163,7 +163,10 @@ class GpuPriorityFusionQueue {
       }
       instructions.push_back(instruction);
     }
-
+    // write here and only read in worker threads
+    for (auto producer : instructions) {
```

Comment:
This loop should run on all fusible instructions in the computation. `instructions` array doesn't have ROOT instructions, because they have no users. Subsequent `ComputeAndSetPriorities` will either crash or return an incorrect result.